### DISCUSSION TITLE
Bare metal: Can Vcpkg extract the .hpp / .cpp files into an external folder structure

### DISCUSSION THREAD

**Discussion by DanL1234** [microsoft/vcpkg#18651] [Q&A]
Hello!

I have a desire to build a single executable for a bare metal project - I'm a way away from chosing a target processor, but think PIC32, for example. Calling "vcpkg integrate install" adds headers and prebuilt libraries to Visual Studio making it very easy to access the dependencies. This will sort out the initial "try it on windows" for ease of development purposes.

However, I will at some point want to pull all source & header files into a single repository/directory structure for another microprocessor specific IDE such as PIC's MPLAB. To do this, at the moment the only options I have are:

> piece-wise copy-pasting the (\buildtrees') source folders
> piece-wise copy-pasting the header files (\installed)
> ensuring correct directory setup/arrangement so files can be found relative
> setting IDE includes & source directories (or environment variables)

This is very laborious.

Can anyone suggest a better solution? 

(Eg, "PS> .\vcpkg extract source & headers ForPackage:Triplet --recurse <directory>")

**Comment by mathisloge** [comment]
vcpkg has a export command: 
Example:
  `vcpkg export zlib:x64-windows`
you can specify the export packaging with one of the options `--raw --nuget --ifw --zip --7zip --chocolatey --prefab`
